src/components/settings/ConfigDatetime.js

```diff
@@ -0,0 +1,92 @@
+import React, { useState } from "react";
+import {
+  closestCenter,
+  DndContext,
+  DragOverlay,
+  KeyboardSensor,
+  PointerSensor,
+  useSensor,
+  useSensors,
+} from "@dnd-kit/core";
+import {
+  arrayMove,
+  SortableContext,
+  sortableKeyboardCoordinates,
+  verticalListSortingStrategy,
+} from "@dnd-kit/sortable";
+
+import { SortableItem } from "./SortableItem";
+import { useTranslation } from "react-i18next";
+import { Header, Button } from "semantic-ui-react";
+
+export function ConfigDatetime() {
+  const [items, setItems] = useState([
+    {
+      id: "1",
+      name: "DateTimeOriginal",
+      rule_type: "exif",
+      exif_tag: "EXIF:DateTimeOriginal",
+      transform_tz: 1,
+      source_tz: "utc",
+      report_tz: "gps_timezonefinder",
+    },
+    {
+      id: "2",
+      name: "QuickTime:CreateDate",
+      rule_type: "exif",
+      exif_tag: "QuickTime:CreateDate",
+      transform_tz: 1,
+      source_tz: "utc",
+      report_tz: "name:Europe/Moscow",
+    },
+    {
+      id: "3",
+      name: "Guess Filename",
+      rule_type: "filesystem",
+      file_property: "mtime",
+      transform_tz: 1,
+      source_tz: "utc",
+      report_tz: "gps_timezonefinder",
+    },
+  ]);
+  const sensors = useSensors(
+    useSensor(PointerSensor),
+    useSensor(KeyboardSensor, {
+      coordinateGetter: sortableKeyboardCoordinates,
+    })
+  );
+  const { t } = useTranslation();
+
+  return (
+    <div style={{ marginTop: 10 }}>
+      <Header as="h3">{t("settings.configdatetime")}</Header>
+      <Button color="green" style={{ marginBottom: 10 }}>
+        Add Rule{" "}
```

Comment:
I guess one thing we can do here is to provide a list of pre-populated elements for users to choose from. Later if we want to go with the power user version we could add expandable section "add advanced rule" or something like that.

Reply:
I added a modal, where you can select a rule:
https://youtu.be/BSosSW9lkU8?t=2665

Should be very expandable, if we add advanced rules. I guess I need now a list of default rules.

Reply:
Thank you for working on this. Looks very nice, and indeed allows future improvements. I have 1 suggestion and 1 concern about the current state.

1. The suggestion is to make the list of pre-defined rules to be delivered to frontend from the backend (by a backend API call). The idea is that if we come up with new types of rules and such we should be able to add it and update the modal-chose-rule list by just one repo change (rather than requiring to make a change to both backend and frontend). I also can imagine a situation when we would want to actually make the pre-defined list dynamic per user (though that part is not very clear to me, but e.g. we could suggest some conditions-enhanced rules using exif values we parsed from the actual user's data, but there are some caveats there).

2. The concern is about the `id` field - I imagine it will make it more difficult to do customization for the rules - maybe we should just drop the `id` field from the item JSONs? I know they are used now for the JS purpose of processing drag events and stuff - so maybe we can just have the logic of `idOfAddedItem := max(existingIds) + 1` (the id of HTML tag or whatever it is, not JSON value) or something like that and don't use it for filtering rules (effectively allowing duplicates)? Or maybe do JSON comparison for equality (which doesn't have `id` field) to determine if the rule to be added is a duplicate.

As to the rules I think we could start with the following:

Default:
  - DateTimeOriginal
  - VideoCreateDateTime_WithGpsTimezone
  - GPSDateTime_WithGpsTimezone
  - Filename
  - VideoCreateDateTime_InUTC

And additional to choose (but not default):
  - FilesystemModifiedTime
  - FilesystemCreateTime
  - GPSDateTime_InUTC (not sure if this one makes sense though)

Reply:
1. Yes, that's a good idea, that we fetch the default rules from the backend. I think it would also make sense to fetch the current configuration from the user. Do you want to work on that part, or should I add that?

2. The id is just any unique identifier. We could use a UUID or the id from the database as a unique identifier. I think that this field makes comparisons easier. I think there is confusion on what this field is supposed to do, because of my copied rules 😅

Reply:
I can work on 1 or you could implement the API part. In any case I'll be only able to continue this next week. If you start working on it please let me know in the discord or here. Or if I start first I'll write here.

Meanwhile I found some tags that might be useful like RIFF:DateTimeOriginal . The bad thing about this is that people don't know which tags are present in their data...

For 2 it just feels like having id for rules doesn't bring benefit (except check for duplicates, but we could do that without id too), but makes things more difficult for the potential new features. Like if we construct new rules on frontend they wouldn't get backend db id, or same rule would have different id. Feels like an extra entity to worry about so maybe better to just not have it.

Reply:
Alright, I think I will then continue working on the frontend part for now, removing the ID's and then adding the API's when they are ready 😄 

RIFF sounds interesting. It's probably an old format, but could be useful.

I can work on adding more EXIF tags 👍